src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c

```diff
@@ -409,21 +409,33 @@ static void init_resp(struct xgq_com_queue_entry *resp, u16 cid, u32 rcode)
 	memset(resp, 0, sizeof(*resp));
 	resp->hdr.cid = cid;
 	resp->hdr.cstate = XGQ_CMD_STATE_COMPLETED;
+	resp->result = KDS_COMPLETED;
+	resp->rcode = rcode;
+}
+
+static void init_resp_abort(struct xgq_com_queue_entry *resp, u16 cid, u32 rcode, int status)
+{
+	memset(resp, 0, sizeof(*resp));
+	resp->hdr.cid = cid;
+	resp->hdr.cstate = XGQ_CMD_STATE_ABORTED;
+	resp->result = status;
 	resp->rcode = rcode;
 }
 
 static inline void
-zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, int ret)
+zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, int ret, int status)
```

Comment:
Should status be an enumeration type.

Reply:
Will update in a separate kds_status change PR